---
title: "Figures"
author: "LD"
date: '2023-06-06'
output: html_document
editor_options: 
  chunk_output_type: console
chunk_output_type: console
---
```{r setup}
library(Seurat)
library(dplyr)
library(tibble)
library(stringr)
library(ggplot2)
library(cowplot)
library(ggbreak) 
library(tidyr)
library(patchwork)
library(ComplexUpset)
library(ggrepel)
library(Polychrome)
library(ComplexHeatmap)
library(gplots)
library(RColorBrewer)
library(randomcoloR)

```

ln -s /home/common/data/output/projects/ENDO/E044/A033/EndoAtlas_GEO_meta_complete_downsampled.rds EndoAtlas.rds
EndoAtlas_meta <- readRDS(paste0(dir_data,"EndoAtlas_meta.rds"))
EndoAtlas <- readRDS(paste0(dir_data,"EndoAtlas.rds"))
EndoAtlas <- AddMetaData(EndoAtlas, EndoAtlas_meta)
saveRDS(EndoAtlas,paste0(dir_data,"EndoAtlas.rds") )

Add additional data to the Seurat metadata and reformat
```{r plotting of cell types per sample}
#Set directory links and create directories
dir_ENDO <- "/home/duempelmann/analysis-projects/ENDO/E044/A085/endometriosis_endometrium_scRNA_atlas/"

dir_data <- paste0(dir_ENDO,"_Data/")
dir_out <- paste0(dir_ENDO,"Figures_out/")
if (!dir.exists(dir_out)) {
  dir.create(dir_out, recursive = TRUE)
}

#Load data
EndoAtlas <- readRDS(paste0(dir_data, "EndoAtlas.rds"))
EndoAtlas_meta <- EndoAtlas@meta.data

#Preserve cell names
EndoAtlas_meta$cell_names <- rownames(EndoAtlas_meta)

##Modify menstrual cycle information
# Create column MenstrualCyclePhase_main based on MenstrualCyclePhase by merging the secretory sub-phases from 
EndoAtlas_meta$MenstrualCyclePhase_main <- as.character(EndoAtlas_meta$MenstrualCyclePhase)
EndoAtlas_meta$MenstrualCyclePhase_main[EndoAtlas_meta$MenstrualCyclePhase_main %in% c("secretory_early", "secretory_mid", "secretory_late")] <- "secretory"
#Factor samples by MenstrualCyclePhase_main for plotting
EndoAtlas_meta$MenstrualCyclePhase_main <- factor(EndoAtlas_meta$MenstrualCyclePhase_main, levels = c("proliferative","periovulatory","secretory"))

##Add additional metadata from manuscript Supplementary Table 1
# Read the tab-delimited text file
SupplementaryTable1 <- read.delim(paste0(dir_data, "SupplementaryTable1.txt"), header = TRUE, stringsAsFactors = FALSE)
# Perform a left join of the `meta` dataframe with `SupplementaryTable1`
EndoAtlas_meta <- dplyr::left_join(
  EndoAtlas_meta,
  SupplementaryTable1 %>% 
    dplyr::select(
      Patient.sample.ID,
      epithelial.pseudotime,
      sequencing.batch,
      X10x.capturing.batch,
      Minor.exclusion.criteria.logical,
      Menstrual.cycle.phase.marker.analysis,
      Cell.type.frequency.analysis..Figure.2e.f.,
      DEG.analysis..Figure.3a.b.,
      Ligand.receptor.analysis..Figure.3c.h.
    ),
  by = c("sample" = "Patient.sample.ID")
)

##Determine order of samples by MenstrualCyclePhase and epithelial.pseudotime
MenstrualCyclePhase_sampleOrder <- EndoAtlas_meta%>%
  select(sample, MenstrualCyclePhase,epithelial.pseudotime) %>%
  distinct() %>%
  arrange(MenstrualCyclePhase, epithelial.pseudotime) %>%
  mutate(MenstrualCyclePhase_sampleOrder = row_number()) %>%
  ungroup()
#Factor samples by MenstrualCyclePhase and epithelial.pseudotime for plotting
EndoAtlas_meta$sample <- factor(EndoAtlas_meta$sample, levels = c(MenstrualCyclePhase_sampleOrder$sample))

#Add cell names as row names
rownames(EndoAtlas_meta) <- EndoAtlas_meta$cell_names

#Save EndoAtlas_meta as .rds
saveRDS(EndoAtlas_meta, paste0(dir_data, "EndoAtlas_meta.rds"))

#EndoAtlas_meta <- readRDS(paste0(dir_data, "EndoAtlas_meta.rds"))

#Save color pals
qual_col_pals <- brewer.pal.info[brewer.pal.info$category == "qual", ]
col_vector <- unlist(mapply(brewer.pal, qual_col_pals$maxcolors, rownames(qual_col_pals)))
```

Save epithelial object and upload to GEO.
```{r}
#Load monocl object
data_monocl <- readRDS("/home/common/data/output/projects/ENDO/E044/A062/trajectory_day5-6-7_greenCircle/trajectory_timepoint0.rds")
listData <- (data_monocl@colData@listData)
listData <- listData["cell_ident"]

EndoAtlas_meta <- readRDS(paste0(dir_data, "EndoAtlas_meta.rds"))
colnames(EndoAtlas_meta)

########################
# Extract cell.ident from listData
cell_ids <- listData[["cell_ident"]]

# Ensure cell_ids is not empty
if (length(cell_ids) == 0) {
  stop("cell.ident in listData is empty. Ensure it is populated before proceeding.")
}

# Filter and order EndoAtlas_meta by cell_ids
filtered_meta <- EndoAtlas_meta[EndoAtlas_meta$cell_names %in% cell_ids, ]
ordered_meta <- filtered_meta[match(cell_ids, filtered_meta$cell_names), ]

# Verify alignment
if (!all(ordered_meta$cell_names == cell_ids)) {
  stop("Ordering of filtered EndoAtlas_meta does not match cell_ids.")
}

# Add columns from ordered_meta to listData
columns_to_add <- c("sample", "EndometriosisStatus", "EndometriosisGrade", "CycleDay", 
                    "ProgesteroneSerum", "MenstrualCyclePhase", "AnnotationMain", 
                    "AnnotationRefined", "cell_names", "MenstrualCyclePhase_main")

for (col in columns_to_add) {
  if (col %in% colnames(ordered_meta)) {
    listData[[col]] <- ordered_meta[[col]]
  } else {
    warning(paste("Column", col, "not found in EndoAtlas_meta. Skipping."))
  }
}

# Update the main object
data_monocl@colData@listData <- listData

data_monocl@principal_graph_aux@listData[["UMAP"]][["pseudotime"]]

saveRDS(data_monocl, paste0(dir_data, "epithelial_cells_monocle3.rds"))
```


###########################################
Main Figures
###########################################

Figure 1c
CODE OK
```{r}

order <- EndoAtlas_meta %>%
  select(AnnotationMain, AnnotationRefined) %>%
  filter(!is.na(AnnotationMain)) %>%
  distinct() %>%
  arrange(AnnotationRefined) %>%
  mutate(AnnotationMain = factor(AnnotationMain, levels = c( "endothelial","stromal","lymphocyte","epithelial","myeloid"))) %>%
  arrange(AnnotationMain)
EndoAtlas$AnnotationRefined <- factor(x = EndoAtlas$AnnotationRefined, levels = order$AnnotationRefined)
Fig1c <- DimPlot(EndoAtlas, reduction = "umap", group.by = "AnnotationRefined",  raster = FALSE)
Fig1c
ggsave(filename = paste0(dir_out, 'Fig1c.pdf'),
       plot = Fig1c,
       width =20, height=9)

```

Figure 1d
CODE OK
```{r plotting of cell types per sample}
#Fig1d_1
Fig1d_1 <- EndoAtlas_meta %>%
  group_by(AnnotationMain) %>%
  summarise(row_count = n()) %>% 
  ggplot(aes(x = AnnotationMain, y = row_count)) +
  geom_col() +
  coord_flip()
Fig1d_1
ggsave(paste0(dir_out, "Fig1d_1.pdf"),
  plot = last_plot(),
  width = 15, height = 6)

#Fig1d_2
Fig1d_2 <- EndoAtlas_meta %>%
  group_by(sample, AnnotationMain) %>%
  summarise(row_count = n()) %>%
  ggplot(aes(x = AnnotationMain, y = row_count)) +
  geom_boxplot() +
  scale_y_log10() +
  coord_flip() +
  theme(axis.text.x = element_text(angle = 90, hjust = 1))
Fig1d_2
ggsave(paste0(dir_out, "Fig1d_2.pdf"),
         plot = last_plot(),
         width = 15, height = 6)

#Fig1d_3 
Fig1d_3 <- EndoAtlas_meta%>%
  filter(!is.na(AnnotationMain)) %>%
  group_by(AnnotationMain, EndometriosisGrade) %>%
  summarise(row_count = n()) %>%
  ggplot(aes(x = AnnotationMain, y = row_count, fill = EndometriosisGrade)) +
  geom_bar(stat = 'identity', position = 'fill') +
  scale_y_continuous(expand = c(0, 0)) +
  coord_flip()
Fig1d_3
ggsave(paste0(dir_out, "Fig1d_3.pdf"),
  plot = last_plot(),
  width = 6, height = 4)

```

Figure 1e, 2a, S3a
CODE running, need to extend the PCA analysis folder
dir_out <- "/home/common/data/output/projects/ENDO/E044/A072/allcells_EndoAtlas_SCTassay/"

```{r}
p <- readRDS(paste0(dir_ENDO,"_Data/03_PCA_analysis/p_EndoAtlas.rds"))
#Figure 1e
pdf(paste0(dir_out,"eigencorplot_final_BH.pdf"), width = 10, height = 5)
eigencorplot(p,
    metavars = c("sample","EndometriosisStatus","EndometriosisGrade","CycleDay","ProgesteroneSerum","MenstrualCyclePhase","MenstrualCyclePhase_main","sequencing.batch", "X10x.capturing.batch"),
    col = c('darkorange3','darkorange', 'white','cadetblue2','cadetblue'),
    corMultipleTestCorrection = "BH"
)
dev.off()

#Figure 2a
biplot(p,
    colby = 'MenstrualCyclePhase',
    # ellipse config
      ellipse = TRUE,
      ellipseType = 't',
      ellipseLevel = 0.90,
      ellipseFill = TRUE,
      ellipseAlpha = 1/4,
      ellipseLineSize = 1.0,
    labSize = 0,drawConnectors = FALSE,
    legendPosition = 'top', legendLabSize = 16, legendIconSize = 8.0)

#Supplementary Data Figure 3a
biplot(p,
    colby = 'EndometriosisStatus',
    labSize = 0,drawConnectors = FALSE, 
    hline = 0, vline = 0,
    legendPosition = 'right')
```

Figure 2b
CODE OK
```{r}
#Figure 2b entire atlas
Fig2b_1 <- DimPlot(EndoAtlas, reduction = "umap", group.by = "MenstrualCyclePhase", raster = FALSE)
Fig2b_1
ggsave(filename = paste0(dir_out, 'Fig2b_1.pdf'),
       plot = Fig2b_1,
       width =20, height=9)

#Figure 2b epithelial cells
Fig2b_2 <- DimPlot(EndoAtlas_epithelial_reintegrated, reduction = "umap", group.by = "MenstrualCyclePhase", raster = FALSE)
Fig2b_2
ggsave(filename = paste0(dir_out, 'Fig2b_2.pdf'),
       plot = Fig2b_2,
       width =20, height=9)

```

Figure 2c
Look in IBU folder for object with trajectory line or A063/A066
```{r}
#Figure 2c epithelial cells
epithelial_cells_monocle3 <- readRDS(paste0(dir_data, "epithelial_cells_monocle3.rds"))

#Pseudotime
Fig2c <- plot_cells(epithelial_cells_monocle3, color_cells_by = "pseudotime",
                     label_cell_groups=FALSE, label_leaves=FALSE, label_branch_points=FALSE,
                     graph_label_size=3, cell_size = 1)
Fig2c
ggsave(filename = paste0(dir_out, 'Fig2c.pdf'),
       plot = Fig2c,
       width =20, height=9)
```

Figure 2d
CODE OK
```{r}
#Remove samples 65 and 105, since these samples are just in between menstrual cycle phases and cannot be clearly assigned
Idents(EndoAtlas) <- EndoAtlas$sample
EndoAtlas_filtereds65s105 <- subset(EndoAtlas, subset = sample %in% setdiff(unique(EndoAtlas$sample), c("sample65", "sample105")))
Idents(EndoAtlas_filtereds65s105) <- EndoAtlas_filtereds65s105$AnnotationMain

#### Plot heatmaps for main cell type specific markers for the different menstrual cycle phases. Compare also Wang et al 2020 marker genes (Wang et al. 2020, DOI: 10.1038/s41591-020-1040-z)

###########################
### main epithelial marker genes for menstrual cycle phases

# Subset celltype of interest
EndoAtlas_filtereds65s105 <- subset(EndoAtlas_filtereds65s105, idents = epithelial)
#main epithelial marker genes for menstrual cycle phases
epithelial_main <- c("SFRP4", "LAMC2", "MMP11", "MMP7", "SPRY1", "TFPI2","STEAP4","ADAMTS8","CAPN6","SLC37A2","SCGB1D2", "MT1F", "MT1X","RIMKLB", "S100P", "DEFB1","G0S2","NNMT","GPX3", "PAEP")

# Do MultiBarHeatmap
DoMultiBarHeatmap(object = data, features = epithelial_main, group.by="MenstrualCyclePhase", additional.group.by="epithelial.pseudotime")
# Save MultiBarHeatmap
ggsave(
    filename = paste0(dir_out, epithelial, "Fig2d_epithelial_DoMultiBarHeatmap.pdf"),
    plot = last_plot(),
    width = 15,
    height = 8)

##Plot our marker genes and Wang et al. 2020 marker genes (Wang et al. 2020, DOI: 10.1038/s41591-020-1040-z)
#Wang et al. 2020 menstrual phase marker genes
Wang_epithelial_mens_markers <- c("PLAU", "MMP7", "THBS1", "CADM1", "NPAS3", "ATP1A1", "ANK3", "ALPL", "TRAK1", "SCGB1D2", "MT1F", "MT1X", "MT1E", "MT1G", "CXCL14", "MAOA", "DPP4", "NUPR1", "GPX3", "PAEP")
#Plot first our genes, then Wang genes
heatmap <- DoHeatmap(
  object = data,
  features = c(setdiff(epithelial_main ,Wang_epithelial_mens_markers), Wang_epithelial_mens_markers),
  cells = NULL,
  group.by = "MenstrualCyclePhase",
  group.bar = TRUE)
#heatmap
ggsave(
    filename = paste0(dir_out, epithelial, "DoMultiBarHeatmap_epithelial_OursAndWangMarkers.pdf"),
    plot = last_plot(),
    width = 15,
    height = 8)


###########################
### main stromal marker genes for menstrual cycle phases

# Subset celltype of interest
EndoAtlas_filtereds65s105 <- subset(EndoAtlas_filtereds65s105, idents = stromal)
#main stromal marker genes for menstrual cycle phases
stromal_main <- c("SFRP1", "H1-1", "H3C2", "WNT5A","CILP","BRINP1", "CFD","PTGDS","SLIT3","RPRM", "TIMP3", "IGFBP1","LEFTY2","LRRC15","LUM")
# Do MultiBarHeatmap
DoMultiBarHeatmap(object = data, features = stromal_main, group.by="MenstrualCyclePhase", additional.group.by="epithelial.pseudotime")
# Save MultiBarHeatmap
ggsave(
    filename = paste0(dir_out, stromal, "Fig2d_stromal_DoMultiBarHeatmap.pdf"),
    plot = last_plot(),
    width = 15,
    height = 10)

##Plot our marker genes and Wang et al. 2020 marker genes (Wang et al. 2020, DOI: 10.1038/s41591-020-1040-z)
#Wang et al. 2020 menstrual phase marker genes
Wang_stromal_mens_markers <- c('STC1', 'NFATC2', 'BMP2' ,'PMAIP1' ,'MMP11' ,'SFRP1', 'WNT5A' ,'ZFYVE21' ,'CILP' ,'SLF2' ,'MATN2', 'S100A4' ,'DKK1' ,'CRYAB', 'FOXO1', 'IL15', 'FGF7', 'LMCD1')
#Plot first our genes, then Wang genes
heatmap <- DoHeatmap(
  object = data,
  features = c(setdiff(stromal_main ,Wang_stromal_mens_markers), Wang_stromal_mens_markers),
  cells = NULL,
  group.by = "MenstrualCyclePhase",
  group.bar = TRUE)
#heatmap
ggsave(
    filename = paste0(dir_out, stromal, "DoMultiBarHeatmap_stromal_OursAndWangMarkers.pdf"),
    plot = last_plot(),
    width = 15,
    height = 15)


###########################
### main endothelial, myeloid and lymphocyte marker genes for menstrual cycle phases
endothelial_main <- c("RGS16","DUOX1","CFD","MYC","TIMP3","LEFTY2","IL1B")
myeloid_main <- c("S100A10","TREM2","STXBP2","CCL2","GNLY","CXCL3","CXCL2")
lymphocyte_main <- c("IL7R","DUSP4","MMP26","MT1G","CXCL3","CXCL2","G0S2")

#Make heatmaps in loop
CELLTYPES <- c("endothelial","lymphocyte","myeloid") 
for (CELLTYPE in CELLTYPES) {
  # Subset celltype of interest
  EndoAtlas_filtereds65s105 <- subset(EndoAtlas_filtereds65s105, idents = CELLTYPE)
  # Do MultiBarHeatmap
  DoMultiBarHeatmap(object = EndoAtlas_filtereds65s105, 
                  features = if (CELLTYPE == "endothelial") endothelial_main
                              else if (CELLTYPE == "myeloid") myeloid_main
                              else if (CELLTYPE == "lymphocyte") lymphocyte_main, 
                  group.by="MenstrualCyclePhase", 
                  additional.group.by="epithelial.pseudotime")
  # Save MultiBarHeatmap
  ggsave(
    filename = paste0(dir_out, "Fig2d_", CELLTYPE, "_DoMultiBarHeatmap.pdf"),
    plot = last_plot(),
    width = 15,
    height = 3)

}

###########################
## Additional strong Menstrual cycle phase markers
stromal_extended <- c("H1-1","H1-2","H1-4","H1-5","H2AC13","H2Ac14","H3C2","H4C3","MKI67","TNC", #prolif/periov
                      "CALB2", "PENK","POSTN",#periovulatory
                      "CAPN6", "CYP26A1","HGD","HLA-DMB","MMP26","UPK1B","TFPI2","ENPP3","PKHD1L1","CILP","CHODL","GDF15","SCGB1D2","SCGB1D4","SCGB2A1","CERNA2","CD9",#secretory_early
                      "MT1E","MT1F","MT1G","MT1H","MT1M", "LINC01320",#secretory_early/_mid
                      "ACKR1", #secretory_mid
                      "BRINP1","C1QC","CCN3","CFD","GPX3","MPTL2","MUC16","MYC","PAEP","PDK4","PTGDS","RPRM","SLIT3","SLPI","SPP1","TGM2","TIMP3","TM1E","TM4SF1",
                      "IGFBP1","IGFBP6","KRT7","LEFTY2","LMCD1","LRRC15","LUM","NID2","PLAGL1","PLAT","PRLR","RBP4","BAMBI","CAB39L","COMP", #secretory_late
                      "APOD","CLDN4","CLU","CP","CXCL13","CXCL14","DEFB1","DEPP1","DKK1") #secretory_early/_mid/_late
epithelial_extended <- unique(c("SFRP4", "LAMC2", "MMP11", "TMEM107", "C12orf75", "UBE2C", "CENPW", "H2AC11", "H3C3", "H2AC13", "CSF3", "CENPE","SPRY1", "TFPI2", "STEAP4", "CKB", "LRRC26", "SNX30", "ADAMTS8", "CAPN6", "FXYD4", "XDH","CCN3", "SLC37A2","MMP26", "SCGB1D2", "SCGB1D4", "CYP26A1", "ENPP3", "PTGDS",  "PLEKHH2", "ZBTB16", "STC1", "IGFBP5", "PDGFRA", "REV3L", "HSPA1A",   "MT1F", "MT1G", "PAEP", "RIMKLB", "S100P", "DEFB1", "G0S2", "GDF15", "C12orf75", "DEFB1", "SPP1", "DKK1", "NNMT", "RBP4"))
endothelial_extended <- c("POSTN","RGS16","TFPI2", "SCGB1D4",  "CYP26A1","MT1G", "CFD", "FABP5","APLN","TIMP3","MYC","MUC16", "NAPSB", "LEFTY2","IL1B", "COMP")
myeloid_extended <- c("TREM2",  "PHYHIPL", "PAEP", "RIMKLB", "LEFTY2", "IL2RB", "APOD", "NNMT",  "STXBP2","GNLY", "S100A10", "OLR1", "MUC16", "CXCL13",  "INSIG1", "GZMA",  "PLAUR",  "MMP26",  "RGCC", "IGSF6",  "C15orf48", "UBD", "CCL19",  "FBP1", "CCL2", "FCN1", "CXCL3", "CORO1A","MT1G","INHBA", "CXCL5", "MMP10","SERPINB2","MMP1", "CCL18","SLC7A5","CCL20","THBS1","AREG", "APOC1","SERPINA1")
lymphocyte_extended <- c("CAPN6","MT1H",  "ENPP3",  "TFPI2", "CLDN3", "DUSP4", "PDK4", "MMP26", "CXCL13", "GIMAP4",  "RIMKLB", "MT1G", "DEFB1", "CFD","IL7R", "LDLRAD4", "LEFTY2", "DOCK10", "CYP26A1", "G0S2","MTND1P23")


```

Figure 2e
CODE OK
```{r plotting of cell types per sample}
##Boxplots of main cell types cell frequency as a percentage of total cells, split by menstrual cycle phase and endometriosis status

#Prepare data for boxplot
df2 <- EndoAtlas_meta%>%
  filter(Minor.exclusion.criteria.logical == "FALSE") %>%
  group_by(sample, AnnotationMain, EndometriosisGrade, MenstrualCyclePhase_main, EndometriosisStatus) %>%
  summarise(row_count = n()) %>%
  ungroup() %>%
  filter(!is.na(AnnotationMain)) %>%
  group_by(sample) %>%
  mutate(AnnotationMain_perc = row_count / sum(row_count) * 100) %>%
  ungroup()

#BoxPlots percentage, EndometriosisStatus 
ggplot(df2, aes(x = MenstrualCyclePhase_main, y = AnnotationMain_perc, fill = EndometriosisStatus)) +
  geom_boxplot(outlier.shape = NA) +
  geom_point(aes(colour = EndometriosisStatus),
             colour="black",pch=21, 
             position = position_jitterdodge()) +
  RotatedAxis() +
  xlab('Sample ID') +
  facet_wrap(~AnnotationMain,scales = "free", ncol = 6)
ggsave(paste0(dir_out, "Fig2e_boxplot.pdf"),
  plot = last_plot(),
  width = 12, height = 4.4)


###Significance testing CTL vs Endo
# Perform t-test CTL vs Endo for each cell type grouped by AnnotationRefined, MenstrualCyclePhase_main
df2_ttest_EndometriosisStatus <- df2 %>%
  group_by(AnnotationMain, MenstrualCyclePhase_main) %>%
  summarise(p.value = t.test(AnnotationMain_perc ~ EndometriosisStatus)$p.value) %>%
  ungroup()
#epithelial, proliferative just not significant (p.value = 0.0898)

###Significance testing prolif vs periov vs sec, CTL + Endo
## Perform t-test Prolif vs Peri for each cell type
df2_ttest_ProlifPeri <- df2 %>%
  dplyr::filter(MenstrualCyclePhase_main %in% c("proliferative","periovulatory")) %>%
  group_by(AnnotationMain) %>%
  summarise(p.value = t.test(AnnotationMain_perc ~ MenstrualCyclePhase_main)$p.value) %>%
  ungroup()
#none significant
## Perform t-test Prolif vs Sec for each cell type
df2_ttest_ProlifSec <- df2 %>%
  dplyr::filter(MenstrualCyclePhase_main %in% c("proliferative","secretory")) %>%
  group_by(AnnotationMain) %>%
  summarise(p.value = t.test(AnnotationMain_perc ~ MenstrualCyclePhase_main)$p.value) %>%
  ungroup()
#epi (0.00977) and str (0.0255) significant

## Perform t-test periovulatory vs sec for each cell type
df2_ttest_PeriSec <- df2 %>%
  dplyr::filter(MenstrualCyclePhase_main %in% c("periovulatory","secretory")) %>%
  group_by(AnnotationMain) %>%
  summarise(p.value = t.test(AnnotationMain_perc ~ MenstrualCyclePhase_main)$p.value) %>%
  ungroup()
#epi (0.00794) and str (0.0137) significant
```


Figure 2f
CODE OK
```{r plotting of cell types per sample}
#Early and Mid Secretory Prv_VSMCs 
Fig2f <- EndoAtlas_meta%>%
  filter(Minor.exclusion.criteria.logical == "FALSE") %>%
  group_by(sample, AnnotationRefined) %>%
  summarise(AnnotationRefined_counts = n()) %>%
  ungroup() %>%
  complete(sample, AnnotationRefined, fill = list(AnnotationRefined_counts = 0)) %>%
  filter(!is.na(AnnotationRefined)) %>%
  group_by(sample) %>%
  mutate(AnnotationRefined_perc = AnnotationRefined_counts / sum(AnnotationRefined_counts) * 100) %>%
  ungroup() %>% #60 samples * 63 cell types = 3780 rows, correct
  filter(!sample %in% c("sample81", "sample87", "sample108", "sample115", "sample119", "sample121"))  #54 samples * 63 cell types = 3402 rows, correct
Fig2f <- left_join(Fig2f, dplyr::distinct(EndoAtlas_meta[,c("sample","EndometriosisStatus","MenstrualCyclePhase_main","MenstrualCyclePhase")]), by = "sample")

secEarylMid_Prv_VSMC_MenstrualCyclePhase_main <- Fig2f %>%
  filter(AnnotationRefined %in% c("Prv_VSMC secretory", "Prv_VSMC secretory_late")) %>% #same result with and without "Prv_VSMC secretory_late"
  dplyr::filter(MenstrualCyclePhase %in% c("proliferative","periovulatory","secretory_mid","secretory_early")) %>%
  group_by(sample, EndometriosisStatus, MenstrualCyclePhase_main) %>%
  summarize(
    total_counts_MenstrualCyclePhase_main = sum(AnnotationRefined_counts, na.rm = TRUE),
    total_perc_MenstrualCyclePhase_main = sum(AnnotationRefined_perc, na.rm = TRUE)
  ) %>%
  ungroup()

ggplot(secEarylMid_Prv_VSMC_MenstrualCyclePhase_main, aes(x = MenstrualCyclePhase_main, y = total_perc_MenstrualCyclePhase_main, fill = EndometriosisStatus)) +
  geom_boxplot(outlier.shape = NA) +
  geom_point(aes(colour = EndometriosisStatus),
             colour="black",pch=21, 
             position = position_jitterdodge(jitter.width = 0.2))
ggsave(paste0(dir_out, "Fig2f_boxplot.pdf"),
  plot = last_plot(),
  width = 17, height = 12)

#Statistical testing
secEarylMid_Prv_VSMC_MenstrualCyclePhase_main_pval <- secEarylMid_Prv_VSMC_MenstrualCyclePhase_main %>%
  dplyr::filter(MenstrualCyclePhase_main %in% c("secretory")) %>% 
  summarise(p.value = t.test(total_perc_MenstrualCyclePhase_main ~ EndometriosisStatus)$p.value) %>%
  ungroup()
#significant, p.value 0.0261

secEarylMid_Prv_VSMC_MenstrualCyclePhase_main_pval_prolif <- secEarylMid_Prv_VSMC_MenstrualCyclePhase_main %>%
  dplyr::filter(MenstrualCyclePhase_main %in% c("proliferative")) %>% 
  summarise(p.value = t.test(total_perc_MenstrualCyclePhase_main ~ EndometriosisStatus)$p.value) %>%
  ungroup()
#not significant, p.value NA

secEarylMid_Prv_VSMC_MenstrualCyclePhase_main_pval_peri <- secEarylMid_Prv_VSMC_MenstrualCyclePhase_main %>%
  dplyr::filter(MenstrualCyclePhase_main %in% c("periovulatory")) %>% 
  summarise(p.value = t.test(total_perc_MenstrualCyclePhase_main ~ EndometriosisStatus)$p.value) %>%
  ungroup()
#not significant, p.value 0.7639853 

#Multiple testing correction
p.adjust(c(0.0261), method = "BH", n = length(3)) #Value does not change
p.adjust(c(0.7639853), method = "BH", n = length(3)) #Value does not change


```


Figure 3a and 3b
```{r}

DEGs
E044A031




```

Figure 3c-h
```{r}

CellChat
E044A076
#cellchat <-readRDS(paste0(dir_out,"cellchat_CntrlvsEndo_Symphony_Refined_Final.rds"))
dir_out <- "/home/common/data/output/projects/ENDO/E044/A076/"

#Figure 3c and Supplementary Figure 5b
gk1 <- rankNet(cellchat, mode = "comparison", stacked = T, do.stat = TRUE)
gk1 
ggsave(plot = gk1,
       filename = paste0(dir_out, 'output_comparison_Refined/rankNet.pdf'), width =10,height=12)


#Figure 3d (or was it done separately? xxx)
setwd(paste0(dir_out,"output_comparison_Refined"))
###circle plot
pathways.show <- c("TNF") 
object.list <- list(Endo = cellchat.Endo, Cntrl = cellchat.Cntrl)
weight.max <- getMaxWeight(object.list, slot.name = c("netP"), attribute = pathways.show) 
par(mfrow = c(1,2), xpd=TRUE)
for (i in 1:length(object.list)) {
  netVisual_aggregate(object.list[[i]], signaling = pathways.show, layout = "circle", edge.weight.max = weight.max[1], edge.width.max = 10, signaling.name = paste(pathways.show, names(object.list)[i]))
}

# Save plot
pdf(file = "EndoCtl_TNF_circle_plot_edge10.pdf")
par(mfrow = c(1, length(object.list)), xpd = TRUE)
for (i in 1:length(object.list)) {
  netVisual_aggregate(object.list[[i]], signaling = pathways.show, layout = "circle", edge.weight.max = weight.max[1], edge.width.max = 10, signaling.name = paste(pathways.show, names(object.list)[i]))
}
dev.off()

# Save plot
pdf(file = "EndoCtl_TNF_circle_plot_edge15.pdf")
par(mfrow = c(1, length(object.list)), xpd = TRUE)
for (i in 1:length(object.list)) {
  netVisual_aggregate(object.list[[i]], signaling = pathways.show, layout = "circle", edge.weight.max = weight.max[1], edge.width.max = 10, signaling.name = paste(pathways.show, names(object.list)[i]))
}
dev.off()
####################


#Load CellChat data
cellchat <- readRDS(paste0(dir_out,"cellchat_CntrlvsEndo_Symphony_Refined_Final.rds"))

#Figure 3e ICAM
netVisual_bubble(cellchat, comparison = c(1, 2),signaling = c("ICAM"), remove.isolate = FALSE) 
ggsave(plot = last_plot(),
       filename = paste0(dir_out, 'output_comparison_Refined/netVisual_bubble_ICAM.pdf'), width =25,height=5)

#Figure 3f CXCL
#Load data
netVisual_bubble(cellchat, comparison = c(1, 2),signaling = c("CXCL"), remove.isolate = FALSE) 
ggsave(plot = last_plot(),
       filename = paste0(dir_out, 'output_comparison_Refined/netVisual_bubble_CXCL.pdf'), width =25,height=7)

#Figure 3g VEGF
netVisual_bubble(cellchat, comparison = c(1, 2),signaling = c("VEGF"), remove.isolate = FALSE) # sources.use = 4, targets.use = c(5:11),
ggsave(plot = last_plot(),
       filename = paste0(dir_out, 'output_comparison_Refined/netVisual_bubble_VEGF.pdf'), width =15,height=5)

#Figure 3h IGFBP
netVisual_bubble(cellchat, comparison = c(1, 2),signaling = c("IGFBP"), remove.isolate = FALSE,targets.use = c(5:11)) 
ggsave(plot = last_plot(),
       filename = paste0(dir_out, 'output_comparison_Refined/netVisual_bubble_IGFBP.pdf'), width =15,height=5)




```



###########################################
Supplementary Data Figures
###########################################


Supplementary Data Figure 1a and 1b
CODE OK
```{r VlnPlots of nCounts and nFeatures}
##VlnPlots for "nFeature_RNA", "nCount_RNA","nCount_SCT","nFeature_SCT"
VlnPlot(EndoAtlas,
  features = c("nFeature_RNA", "nCount_RNA","nCount_SCT","nFeature_SCT"),
  group.by='EndometriosisStatus', ncol = 1, pt.size=0)
ggsave(paste0(dir_out, "VlnPlot_GroupByEndometriosisStatus_nFeature_nCounts_RNA_SCT_percMT.pdf"),
  plot = last_plot(),
  width = 4, height = 13)

```

```{r different counts and means/medians}
#Percentage of main 5
table <- as.data.frame(table(EndoAtlas_meta$AnnotationMain))
table$Percentage <- round((table$Freq / sum(table$Freq)) * 100, 1)

#Number of cells for CTL and Endo
table(EndoAtlas_meta$EndometriosisStatus)

#Number of patients for CTL and Endo
EndoAtlas_meta %>% select(sample,EndometriosisStatus) %>%
  distinct() %>%
  group_by(EndometriosisStatus) %>%
  summarize(number_of_rows = n())

#mean number of cells per sample
mean(table(EndoAtlas_meta$sample))

#median count/feature
median_summary <- EndoAtlas_meta %>%
  select(nCount_RNA, nCount_SCT, nFeature_RNA, nFeature_SCT) %>%
  summarize(
    median_nCount_RNA = median(nCount_RNA),
    median_nCount_SCT = median(nCount_SCT),
    median_nFeature_RNA = median(nFeature_RNA),
    median_nFeature_SCT = median(nFeature_SCT)
  )
```

Supplementary Data Figure 1c
CODE OK
```{r}
SFig1c <- FeaturePlot(EndoAtlas, features = "CycleDay", raster = FALSE)
SFig1c
ggsave(filename = paste0(dir_out, 'SFig1c.pdf'),
       plot = SFig1c,
       width =20, height=9)
```

Supplementary Data Figure 1d
CODE OK
```{r}
SFig1d <- FeaturePlot(EndoAtlas, features = "ProgesteroneSerum", raster = FALSE)
SFig1d
ggsave(filename = paste0(dir_out, 'SFig1d.pdf'),
       plot = SFig1d,
       width =20, height=9)

```

Supplementary Data Figure 1e

```{r}
dir_out <- paste0(dir_ENDO,"_Data/03_PCA_analysis/")

p <- readRDS(paste0(dir_ENDO,"_Data/03_PCA_analysis/p_EndoAtlas_DEGsamples.rds"))

#Figure 1e
pdf(paste0(dir_out,"eigencorplot_final_BH.pdf"), width = 10, height = 5)
eigencorplot(p,
    metavars = c("sample","EndometriosisStatus","EndometriosisGrade","CycleDay","ProgesteroneSerum","MenstrualCyclePhase","MenstrualCyclePhase_main","sequencing.batch", "X10x.capturing.batch"),
    col = c('darkorange3','darkorange', 'white','cadetblue2','cadetblue'),
    corMultipleTestCorrection = "BH"
)
dev.off()
```

Supplementary Data Figure 1f
CODE OK
"/home/common/data/output/projects/ENDO/E044/A082/SCTassay/"
```{r}

#Define subsets of main annotation
subsets <- c("myeloid", "endothelial", "epithelial", "lymphocyte","stromal")

#Plot eigencorplots for the main annotations
plotting_PCA <- function(SUBSET) {
    p <- readRDS(paste0(dir_ENDO,"_Data/03_PCA_analysis/p_EndoAtlas_",SUBSET,".rds"))
      
    pdf(paste0(dir_out,"eigencorplot_final_BH_",SUBSET,".pdf"), width = 10, height = 5)
    eigencorplot(p,
    metavars = c("sample","EndometriosisStatus","EndometriosisGrade","CycleDay","ProgesteroneSerum","MenstrualCyclePhase","MenstrualCyclePhase_main","sequencing.batch", "X10x.capturing.batch"),
    col = c('darkorange3','darkorange', 'white','cadetblue2','cadetblue'),
    corMultipleTestCorrection = "BH"
    )
    dev.off()
}

lapply(subsets, plotting_PCA)
```


Supplementary Data Figure 2
CODE OK
```{refined cluster markers}

# Set SCT as default and remove RNA to save space
DefaultAssay(EndoAtlas) <- "SCT"
EndoAtlas@assays$RNA <- NULL
gc()
# Remove data without annotation (no NAs!)
Idents(EndoAtlas) <- EndoAtlas$AnnotationMain
EndoAtlas <- subset(EndoAtlas, subset = AnnotationMain %in% levels(as.factor(AnnotationMain$Symphony_Refined)))
gc()

### AnnotationMain markers
genes_to_plot <- c("EPCAM","CDH1","PDGFRA","COL1A1", "VWF","PECAM1","PTPRC","CD68","CD14","CD2") #correct xxx
pops_to_plot <- c("stromal","epithelial","myeloid","lymphocyte","endothelial")
# DotPlot
Seurat::DotPlot(
  object = EndoAtlas,
  scale = T,
  features = genes_to_plot,
  idents = pops_to_plot,
) + theme(axis.text.x = element_text(angle = 90))
# Save DotPlot
ggsave(plot = last_plot(), "./SFig2_AnnotationMain.pdf", width = 10, height = 5)


###Set identity to AnnotationRefined
Idents(EndoAtlas) <- EndoAtlas$AnnotationRefined

### DC markers (from Tan2022 Fig. 5a)
genes_to_plot <- c("CD1C", "CLEC9A", "LAMP3")
pops_to_plot <- c("mDC", "cDC1", "DC3", "cDC2", "pre-cDC2")
# reorder clusters
pops_seurat <- subset(EndoAtlas, idents = pops_to_plot)
pops_seurat@active.ident <- factor(pops_seurat@active.ident, 
                            levels=pops_to_plot)
# DotPlot
p <- Seurat::DotPlot(
  object = pops_seurat,
  scale = T,
  features = genes_to_plot,
  idents = pops_to_plot,
) + theme(axis.text.x = element_text(angle = 90))
# Save DotPlot
ggsave(plot = p, "./dotplots/Tan2022Fig3b_DCs_SCT.pdf", width = 5, height = 7)


### Lymphocytes markers (from Tan2022 Fig. 4b)
genes_to_plot <- c("FCGR3A","KLRD1","ITGA1","LILRB1","ITGAX","KLRC1","KLRC2","EPAS1","KIR2DL4","IGFBP2","CD160","GZMH","KLRF1","FGFBP2","TBX21","CD8A","CD8B","CRTAM","CCR5","NCR3","RORC","IL7R","TCF7","CD4","CCR7","SELL","LEF1","FOXP3","CTLA4","IL2RA","CXCR6","ZNF683","ITGAE","XCL1","SLC4A10","DPP4","GZMK","IGKC", "MZB1", "DERL3", "CD79A", "CD79B", "MS4A1", "BANK1")
pops_to_plot <- c(EndoAtlas@meta.data %>% filter(AnnotationMain == "lymphocyte") %>% pull(AnnotationRefined) %>% unique())

# reorder clusters
pops_seurat <- subset(EndoAtlas, idents = pops_to_plot)
pops_seurat@active.ident <- factor(pops_seurat@active.ident, 
                           levels=c("B cell", "plasma", "CD8 T$_{RM}$","CD4 T$_{RM}$","T$_{Reg}$", "T$_N$/T$_{CM}$" , "ILC" , "T$_{EM}$","CTL", "pNK", "NK3","NK2","NK1"))
# DotPlot
p <- Seurat::DotPlot(
  object = pops_seurat,
  scale = T,
  features = genes_to_plot,
  idents = pops_to_plot,
) + theme(axis.text.x = element_text(angle = 90))
# Save DotPlot
ggsave(plot = p, "./dotplots/Tan2022Fig3b_Lymphocytes_SCT.pdf", width = 17, height = 7)

### epithelial markers (from Tan2022 Sup Fig. 7c)
genes_to_plot <- c("TPPP3", "PIFO", "FOXJ1", "TP63", "KRT5", "SOX9", "LGR5", "MUC5B", "TFF3", "RUNX3", "SAA1", "SIX1", "PROM1", "PROM2", "ANPEP", "KIAA1324", "SPDEF", "SMAD9", "CD36", "HSD17B2", "SCGB2A2", "GDA", "FGFR2", "WNT7A", "FGF9", "PTGS1", "MSLN", "UPK3B", "CALB2", "PGR", "ESR1", "MMP7", "TIMP1", "IDO1", "PSAT1", "ENPP3", "GNG11", "CREB3L1", "IHH", "TOP2A", "MT1F", "MT1E", "MT1X", "SERPINA5", "SPP1", "PAEP", "CXCL14", "DPP4")
pops_to_plot <- c("mesothelial", "ciliated", "mid-secretory", "lumenal 2", "lumenal 1", "lumenal", "glandular early-secretory", "glandular", "TP63+/KRT5+", "MUC5B+")
# reorder clusters
pops_seurat <- subset(EndoAtlas, idents = pops_to_plot)
pops_seurat@active.ident <- factor(pops_seurat@active.ident, 
                            levels=pops_to_plot)
#DotPlot
p <- Seurat::DotPlot(
  object = pops_seurat,
  scale = T,
  features = genes_to_plot,
  idents = pops_to_plot,
) + theme(axis.text.x = element_text(angle = 90))
# Save DotPlot
ggsave(plot = p, "./dotplots/Tan2022Fig3b_epithelial_SCT.pdf", width = 17, height = 7)


#myeloid markers ( from Tan2022 Fig. 4b)
genes_to_plot <- c("FLOR2", "LYVE1", "MRC1", "CX3CR1", "ICAM2", "CLEC5A", "CCR2", "VEGFA", "FN1", "SPARC", "APOE", "SPP1")
pops_to_plot <- c(EndoAtlas@meta.data %>% filter(AnnotationMain == "myeloid") %>% pull(AnnotationRefined) %>% unique())
pops_to_plot <- c("M$\\Phi$5-activated","M$\\Phi$4-infiltrated","M$\\Phi$3-APOE","M$\\Phi$2-peritoneal","M$\\Phi$1-LYVE1")
# reorder clusters
pops_seurat <- subset(EndoAtlas, idents = pops_to_plot)
pops_seurat@active.ident <- factor(pops_seurat@active.ident, 
                            levels=pops_to_plot)
# DotPlot
p <- Seurat::DotPlot(
  object = pops_seurat,
  scale = T,
  features = genes_to_plot,
  idents = pops_to_plot,
) + theme(axis.text.x = element_text(angle = 90))
# Save DotPlot
ggsave(plot = p, "./dotplots/Tan2022Fig3b_macrophages_SCT.pdf", width = 10, height = 7)


#Prv and VSMC markers (from Tan2022 Fig. 3b)
genes_to_plot <- c("RERGL", "MCAM", "MYH11", "STEAP4","GGT5", "RGS5")
pops_to_plot <- c("Prv-MYH11", "Prv-CCL19", "Prv-STEAP4", "VSMC")
# reorder clusters
pops_seurat <- subset(EndoAtlas, idents = pops_to_plot)
pops_seurat@active.ident <- factor(pops_seurat@active.ident, 
                            levels=c("VSMC", "Prv-STEAP4", "Prv-CCL19","Prv-MYH11"))
# DotPlot
p <- Seurat::DotPlot(
  object = pops_seurat,
  scale = T,
  features = genes_to_plot,
  idents = pops_to_plot,
) + theme(axis.text.x = element_text(angle = 90))
# Save DotPlot
ggsave(plot = p, "./dotplots/Tan2022Fig3b_stromal_SCT.pdf", width = 7, height = 7)

```

Supplementary Figure 3b and e
CODE running
```{r}
epithelial_cells_monocle3 <- readRDS(paste0(dir_data, "epithelial_cells_monocle3.rds"))

#Menstrual Cycle Day
SFig3b <- plot_cells(epithelial_cells_monocle3, color_cells_by = "CycleDay",
                     label_cell_groups=FALSE, label_leaves=FALSE, label_branch_points=FALSE,
                     graph_label_size=3, cell_size = 1)
SFig3b
ggsave(filename = paste0(dir_out, 'SFig3b.pdf'),
       plot = SFig3b,
       width =20, height=9)


#Menstual Cycle Phase Wang
SFig3e <- plot_cells(epithelial_cells_monocle3 , color_cells_by = "WangMenstrualCyclePhase",
                     label_cell_groups=FALSE, label_leaves=FALSE, label_branch_points=FALSE,
                     graph_label_size=3, cell_size = 1)
SFig3e
ggsave(filename = paste0(dir_out, 'SFig3e.pdf'),
       plot = SFig3e,
       width =20, height=9)


```


Supplementary Data Figure 3c
Look in IBU folder for object with trajectory line or A063/A066
```{r}
#Figure S3c epithelial cells
FeaturePlot
Fig3c <- DimPlot(EndoAtlas_epithelial_reintegrated, reduction = "umap", group.by = "MenstrualCycleDay", raster = FALSE)
Fig3c
ggsave(filename = paste0(dir_out, 'Fig3c.pdf'),
       plot = Fig3c,
       width =20, height=9)
```

Supplementary Data Figure 3f
CODE OK
```{r}
#Figure 2b entire atlas
Fig2b_1 <- DimPlot(EndoAtlas, reduction = "umap", group.by = "Wang2020Phase", raster = FALSE)
Fig2b_1
ggsave(filename = paste0(dir_out, 'Fig2b_1.pdf'),
       plot = Fig2b_1,
       width =20, height=9)

```

Supplementary Data Figure 3 g
CODE OK
```{r}
####### Download data from Tan et al. 2022 (DOI: 10.1038/s41556-022-00961-5) through the command line
#cd endometriosis_endometrium_scRNA_atlas/_Data
#wget --no-check-certificate https://singlecell.jax.org/datasets/endometriosis/h5ad/endo-2022_epithelial.h5ad

####### Convert .h5ad files to .h5seurat files
library(SeuratDisk)
Convert(paste0(dir_data,"endo-2022_epithelial.h5ad"), dest = paste0(dir_data,"endo-2022_epithelial.h5seurat"), assay = "RNA", overwrite = TRUE)
Tan_epithelial <- LoadH5Seurat(paste0(dir_data,"endo-2022_epithelial.h5seurat"), assays = "RNA")

#Create scale.data slot for the RNA assay required for heatmapping
Tan_epithelial_features <- Tan_epithelial@assays[["RNA"]]@data@Dimnames[[1]]
Tan_epithelial <- ScaleData(object = Tan_epithelial, features = Tan_epithelial_features)

#Subset cells from the endometrium (exclude cells from endometriosis lesions)
Idents(Tan_epithelial) <- Tan_epithelial$sample_type_rename
Tan_epithelial <- subset(Tan_epithelial, idents = c("Ctrl", "EuE"))

# factor levels in better order (E10 and E11 don't have endometrium cells)
Tan_epithelial$PID <- factor(Tan_epithelial$PID, levels = c("C01", "C02", "C03", "E01", "E02", "E03", "E04", "E05", "E06", "E07", "E08", "E09"))

# Generate heatmap of the endometrium cells from Tan et al. 2022 with the same epithelial menstrual marker genes as in our manuscript Figure 2d, for comparison
feature_list_figure2 <- c("SFRP4", "LAMC2", "MMP11", "MMP7", "SPRY1", "TFPI2","STEAP4","ADAMTS8","CAPN6","SLC37A2","SCGB1D2", "MT1F", "MT1X","RIMKLB", "S100P", "DEFB1","G0S2","NNMT","GPX3", "PAEP")

heatmap <- DoHeatmap(
  object = Tan_epithelial,
  features = intersect(feature_list_figure2, Tan_epithelial_features),
  cells = NULL,
  group.by = "PID",
  assay = "RNA",
  group.bar = TRUE) + 
  scale_fill_gradientn(limits = c(-2, 2), colors = c("#FF00FF", "#000000", "#FFFF00"))
heatmap
ggsave(
    filename = paste0(dir_out, "Tan_epithelial_Fig2EpiMarkers.pdf"),
    plot = last_plot(),
    width = 12.3,
    height = 5)


#As an alternative control heatmap of menstrual marker genes from Wang et al. 2020 (DOI: 10.1038/s41591-020-1040-z)
feature_list_Wang <- c("PLAU", "MMP7", "THBS1", "CADM1", "NPAS3", "ATP1A1", "ANK3", "ALPL", "TRAK1", "SCGB1D2", "MT1F", "MT1X", "MT1E", "MT1G", "CXCL14", "MAOA", "DPP4", "NUPR1", "GPX3", "PAEP")

heatmap <- DoHeatmap(
  object = Tan_epithelial,
  features = intersect(feature_list_Wang, Tan_epithelial_features),
  cells = NULL,
  group.by = "PID",
  assay = "RNA",
  group.bar = TRUE) + 
  scale_fill_gradientn(limits = c(-2, 2), colors = c("#FF00FF", "#000000", "#FFFF00"))
heatmap
ggsave(
    filename = paste0(dir_out, "Tan_epithelial_WangEpiMarkers.pdf"),
    plot = last_plot(),
    width = 15,
    height = 15)

#The menstrual marker gene expression varies largely between the endometrium samples in the Tan et al. 2022 dataset, strongly indicating not homogeneous menstrual cycle phase between the endometrium samples. 
#The expression distribution of menstrual marker genes from our manuscript figure 2d and Wang et al. 2020 look comparable.

```

Supplementary Data Figure 3 h
CODE OK
```{r}
####### Download data from Tan et al. 2022 (DOI: 10.1038/s41556-022-00961-5) through the command line
#cd endometriosis_endometrium_scRNA_atlas/_Data
#wget --no-check-certificate https://singlecell.jax.org/datasets/endometriosis/h5ad/endo-2022_epithelial.h5ad

####### Convert .h5ad files to .h5seurat files
library(SeuratDisk)
Convert(paste0(dir_data,"endo-2022_stromal.h5ad"), dest = paste0(dir_data,"endo-2022_stromal.h5seurat"), assay = "RNA", overwrite = TRUE)
Tan_stromal <- LoadH5Seurat(paste0(dir_data,"endo-2022_stromal.h5seurat"), assays = "RNA")

#Create scale.data slot for the RNA assay required for heatmapping
Tan_stromal_features <- Tan_stromal@assays[["RNA"]]@data@Dimnames[[1]]
Tan_stromal <- ScaleData(object = Tan_stromal, features = Tan_stromal_features)

#Subset cells from the endometrium (exclude cells from endometriosis lesions)
Idents(Tan_stromal) <- Tan_stromal$sample_type_rename
Tan_stromal <- subset(Tan_stromal, idents = c("Ctrl", "EuE"))

# factor levels in better order (E10 and E11 don't have endometrium cells)
Tan_stromal$PID <- factor(Tan_stromal$PID, levels = c("C01", "C02", "C03", "E01", "E02", "E03", "E04", "E05", "E06", "E07", "E08", "E09"))

# Generate heatmap of the endometrium cells from Tan et al. 2022 with the same epithelial menstrual marker genes as in our manuscript Figure 2d, for comparison
feature_list_figure2_stromal <- c("SFRP1", "H1-1", "H3C2", "WNT5A","CILP","BRINP1", "CFD","PTGDS","SLIT3","RPRM", "TIMP3", "IGFBP1","LEFTY2","LRRC15","LUM")

heatmap <- DoHeatmap(
  object = Tan_stromal,
  features = intersect(feature_list_figure2_stromal, Tan_stromal_features),
  cells = NULL,
  group.by = "PID",
  assay = "RNA",
  group.bar = TRUE) + 
  scale_fill_gradientn(limits = c(-2, 2), colors = c("#FF00FF", "#000000", "#FFFF00"))
heatmap
ggsave(
    filename = paste0(dir_out, "Tan_stromal_Fig2EpiMarkers.pdf"),
    plot = last_plot(),
    width = 12.3,
    height = 5)


#As an alternative control heatmap of menstrual marker genes from Wang et al. 2020 (DOI: 10.1038/s41591-020-1040-z)
feature_list_Wang_stromal <- c('STC1', 'NFATC2', 'BMP2' ,'PMAIP1' ,'MMP11' ,'SFRP1', 'WNT5A' ,'ZFYVE21' ,'CILP' ,'SLF2' ,'MATN2', 'S100A4' ,'DKK1' ,'CRYAB', 'FOXO1', 'IL15', 'FGF7', 'LMCD1')

heatmap <- DoHeatmap(
  object = Tan_stromal,
  features = intersect(feature_list_Wang_stromal, Tan_stromal_features),
  cells = NULL,
  group.by = "PID",
  assay = "RNA",
  group.bar = TRUE) + 
  scale_fill_gradientn(limits = c(-2, 2), colors = c("#FF00FF", "#000000", "#FFFF00"))
heatmap
ggsave(
    filename = paste0(dir_out, "Tan_stromal_WangEpiMarkers.pdf"),
    plot = last_plot(),
    width = 15,
    height = 15)

#The menstrual marker gene expression varies largely between the endometrium samples in the Tan et al. 2022 dataset, strongly indicating not homogeneous menstrual cycle phases between the endometrium samples. 
#The expression distribution of menstrual marker genes from our manuscript figure 2d and Wang et al. 2020 look comparable.

```

Supplementary Data Figure 4a
CODE OK
```{r plotting of cell types per sample}
##Barplots of cell type frequency split by the main 5 cell types

#Set colour palette
P63 = createPalette(63, c("#ff0000", "#00ff00", "#0000ff")) 
swatch(P63)
names(P63) <- NULL


#Make df with number of AnnotationRefined cells in each sample
df <- EndoAtlas_meta%>%
  filter(!is.na(AnnotationMain)) %>%
  group_by(sample, AnnotationMain, AnnotationRefined, EndometriosisGrade, MenstrualCyclePhase, epithelial.pseudotime, EndometriosisStatus) %>%
  summarise(row_count = n()) %>%
  ungroup() 

#Make bar plot
p <- ggplot(df, aes(x = sample, y = row_count, fill = AnnotationRefined)) +
  geom_bar(stat = 'identity', position = 'fill') +
  scale_y_continuous(expand = c(0, 0)) +
  scale_fill_manual(values = sample(P63)) +
  xlab('Sample ID') +
  theme(
    axis.text.x = element_text(angle = 90, hjust = 1),
    panel.grid.minor = element_blank(),
    panel.grid.major.y = element_line(colour = "lightgray", size = 0.5)
  ) +
  facet_grid(AnnotationMain ~ ., scales = "free_y")
p
#Save barplot
ggsave(paste0(dir_out, "SFig4a_barplot.pdf"),
       plot = p,
       width = 15, height = 6)

```


Supplementary Data Figure 4b
CODE OK
```{r plotting of cell types per sample}
##########################################################################################
##Plot differences for samples without minor exclusion criteria

#Make df 
#Introduce 0 for the ones with 0 count!
SFig4b <- EndoAtlas_meta%>%
  filter(Minor.exclusion.criteria.logical == "FALSE") %>%
  group_by(sample, AnnotationRefined) %>%
  summarise(AnnotationRefined_counts = n()) %>%
  ungroup() %>%
  complete(sample, AnnotationRefined, fill = list(AnnotationRefined_counts = 0)) %>%
  filter(!is.na(AnnotationRefined)) %>%
  group_by(sample) %>%
  mutate(AnnotationRefined_perc = AnnotationRefined_counts / sum(AnnotationRefined_counts) * 100) %>%
  ungroup() %>% #60 samples * 63 cell types = 3780 rows, correct
  filter(!sample %in% c("sample81", "sample87", "sample108", "sample115", "sample119", "sample121"))  #54 samples * 63 cell types = 3402 rows, correct
SFig4b <- left_join(SFig4b, dplyr::distinct(EndoAtlas_meta[,c("sample","EndometriosisStatus","MenstrualCyclePhase_main")]), by = "sample")
write.csv(SFig4b, paste0(dir_out,"SFig4b_AnnotationRefined_frequency.csv"))

#BoxPlots percentage, EndometriosisStatus, percentage ###for paper Extended Data Fig. S4
ggplot(SFig4b, aes(x = MenstrualCyclePhase_main, y = AnnotationRefined_perc, fill = EndometriosisStatus)) +
  geom_boxplot(outlier.shape = NA) +
  geom_point(aes(colour = EndometriosisStatus),
             colour="black",pch=21, 
             position = position_jitterdodge(jitter.width = 0.2)) +
  facet_wrap(~AnnotationRefined, scales = "free", ncol = 7) +
  theme_bw() +
  expand_limits(y = 0)
  
ggsave(paste0(dir_out, "SFig4b_EndometriosisStatus_perc.pdf"),
  plot = last_plot(),
  width = 20, height = 12)

## Perform t-test Prolif vs Sec for each cell type
SFig4b_ttest_ProlifSec <- SFig4b %>%
  dplyr::filter(MenstrualCyclePhase_main %in% c("proliferative","secretory")) %>%
  group_by(AnnotationRefined) %>%
  summarise(p.value = t.test(AnnotationRefined_perc ~ MenstrualCyclePhase_main)$p.value) %>%
  ungroup()
SFig4b_ttest_ProlifSec$p.value_adjBH <- p.adjust(SFig4b_ttest_ProlifSec$p.value, method = "BH", n = length(SFig4b_ttest_ProlifSec$p.value))
write.csv(SFig4b_ttest_ProlifSec, paste0(dir_out,"SFig4b_ttest_ProlifSec_adjpval.csv"))

# Perform t-test CTL vs Endo for each cell type grouped by AnnotationRefined, MenstrualCyclePhase_main
SFig4b_ttest_EndometriosisStatus <- SFig4b %>%
  group_by(AnnotationRefined, MenstrualCyclePhase_main) %>%
  summarise(p.value = t.test(AnnotationRefined_perc ~ EndometriosisStatus)$p.value) %>%
  ungroup()
#Prv_VSMC secretory pval 0.009075467
SFig4b_ttest_EndometriosisStatus$p.value_adjBH <- p.adjust(SFig4b_ttest_EndometriosisStatus$p.value, method = "BH", n = length(SFig4b_ttest_EndometriosisStatus$p.value))
write.csv(SFig4b_ttest_EndometriosisStatus, paste0(dir_out,"SFig4b_ttest_EndometriosisStatus_adjpval.csv"))
#After correction for multiple testing Prv_VSMC secretory is not significant anymore

```

Supplementary Data Figure 5a
```{r}
DEGs





```

Supplementary Data Figure 5b-d
```{r}

#Figure 3c and Supplementary Figure 5b
gk1 <- rankNet(cellchat, mode = "comparison", stacked = T, do.stat = TRUE)
gk1 
ggsave(plot = gk1,
       filename = paste0(dir_out, 'output_comparison_Refined/rankNet.pdf'), width =10,height=12)


#Load CellChat data
cellchat <- readRDS(paste0(dir_out,"cellchat_CntrlvsEndo_Symphony_Refined_Final.rds"))

#Supplementary Data Figure 5d TNF
netVisual_bubble(cellchat, comparison = c(1, 2),signaling = c("TNF"), remove.isolate = FALSE) 
ggsave(plot = last_plot(),
       filename = paste0(dir_out, 'output_comparison_Refined/netVisual_bubble_TNF.pdf'), width =25,height=5)


```

#Supplementary Data Figure 6a, b
#Need to change path
```{r barplot distribution of cv splits, all cells, prolif}
meta <- read.csv("/home/common/data/output/projects/ENDO/E042/A001/summary_plots_cv_splits/metadata_prolif_eval_fold_all.csv")

library(reshape2)
library(ggplot2)
library(forcats)

#Supplementary Data Figure 6a
# reshape df
melted_meta <- melt(meta, id.vars = c("lane", "endo_grade"), measure.vars = c("fold1", "fold2", "fold3", "fold4", "fold5"))
## create the bar plot for endo_EndometriosisGrade
SFig6a <- ggplot(melted_meta, aes(x = variable, fill = endo_grade)) +
  geom_bar(position = "stack", width = 0.7) +
  facet_wrap(~ fct_rev(value), ncol = 2) +
  labs(title = "Bar Plot of Samples by Endo EndometriosisGrade",
       x = "Fold",
       y = "Number of Samples") +
  scale_fill_manual(values = c("severe" = "red", "mild" = "orange", "CTL" = "blue")) +
  theme_minimal() +
  coord_flip()
SFig6a
ggsave(paste0(dir_out,"SFig6a_barplot.pdf"), plot = SFig6a, height = 6.3, width = 7.2)

#Supplementary Data Figure 6b
SFig6b <- ggplot(melted_meta, aes(x = variable, fill = lane)) +
  geom_bar(position = "stack", width = 0.7) +
  facet_wrap(~ fct_rev(value), ncol = 2) +
  labs(title = "Bar Plot of Samples by lane",
       x = "Fold",
       y = "Number of Samples") +
  #scale_fill_manual(values = c("severe" = "red", "mild" = "orange", "CTL" = "blue")) +
  theme_minimal() +
  coord_flip()
SFig6b
ggsave(paste0(dir_out,"SFig6b_barplot.pdf"), plot = SFig6b, height = 6.3, width = 7.2)

```

Supplementary Data Figure 6c
#Need to change path
```{r}

metadata <- read.csv('/home/common/data/output/projects/ENDO/E042/A001/summary_plots_cv_splits/metadata_prolif_eval_fold_all.csv')
metadata$new_column <- NULL
metadata$batch <- NULL


# Remove 'names' column as it will be used as row labels
row_names <- metadata$names
metadata <- metadata[, -1]

# Convert non-numeric columns to factors
metadata[, c("EndometriosisStatus", "lane",  "endo_EndometriosisGrade",  "fold1", "fold2", "fold3", "fold4", "fold5")] <- lapply(metadata[, c("EndometriosisStatus", "lane", "endo_EndometriosisGrade", "fold1", "fold2", "fold3", "fold4", "fold5")], as.factor)



# Create heatmap using ComplexHeatmap
pdf(paste0(dir_out, "metadata_prolif_eval_fold_all_heatmap.pdf"), width = 7, height = 9)
Heatmap(metadata,
        name = "Metadata",
        row_names_side = "left",
        row_names_gp = gpar(fontsize = 8),
        cluster_rows = FALSE,
        cluster_columns = FALSE,
        show_column_names = TRUE,
        show_row_names = TRUE,
        column_title = "Columns",
        row_title = "Row Names",
        heatmap_legend_param = list(title = "Legend")
)
dev.off()

```


##################################################################################################################
Scailyte part, extra code to delete

#Supplementary Data Figure 6a, b
```{r barplot distribution of cv splits, all cells, prolif}
meta <- read.csv("/home/common/data/output/projects/ENDO/E042/A001/summary_plots_cv_splits/metadata_prolif_eval_fold_all.csv")

library(reshape2)
library(ggplot2)
library(forcats)

#Supplementary Data Figure 6a
# reshape df
melted_meta <- melt(meta, id.vars = c("lane", "endo_grade"), measure.vars = c("fold1", "fold2", "fold3", "fold4", "fold5"))
## create the bar plot for endo_EndometriosisGrade
ggplot(melted_meta, aes(x = variable, fill = endo_grade)) +
  geom_bar(position = "stack", width = 0.7) +
  facet_wrap(~ fct_rev(value), ncol = 2) +
  labs(title = "Bar Plot of Samples by Endo EndometriosisGrade",
       x = "Fold",
       y = "Number of Samples") +
  scale_fill_manual(values = c("severe" = "red", "mild" = "orange", "CTL" = "blue")) +
  theme_minimal() +
  coord_flip()
ggsave(paste0(dir_out,"Bar Plot of Samples by Endo EndometriosisGrade_allcells.pdf"), plot = last_plot(), height = 6.3, width = 7.2)

#Supplementary Data Figure 6b
ggplot(melted_meta, aes(x = variable, fill = lane)) +
  geom_bar(position = "stack", width = 0.7) +
  facet_wrap(~ fct_rev(value), ncol = 2) +
  labs(title = "Bar Plot of Samples by lane",
       x = "Fold",
       y = "Number of Samples") +
  #scale_fill_manual(values = c("severe" = "red", "mild" = "orange", "CTL" = "blue")) +
  theme_minimal() +
  coord_flip()
ggsave(paste0(dir_out,"Bar Plot of Samples by lane_allcells.pdf"), plot = last_plot(), height = 6.3, width = 7.2)

#Supplementary Data Figure 6e
##Upsetplot of eval samples
df <- meta %>%
  rownames_to_column(var = "name") %>%
  select(-name) %>% 
  select(fold1, fold2, fold3, fold4, fold5) %>%
  mutate_all(~ ifelse(. == 'eval', TRUE, FALSE)) %>%
  filter(rowSums(. == TRUE) > 0)
pdf(file.path(dir_out, "upset_CellCnnsig_all_evalsamples.pdf"))
upset(df,
  colnames(df)) +
  ggtitle("all cells")
dev.off()


```

```{r Plot DEGs from Tan proliferative, vulcano and DotPlot}
#RNA assay DEGs from proliferative samples
RNAlistRDS005_df <- read.csv("/home/common/data/output/projects/ENDO/E044/A031/RNAlistRDS005_unmerged.csv") %>%
  filter(annotation_reference == "Tan") %>%
  filter(AveExpr > 18.53202)
##Exclude Haemoglobin genes, mitochondrial genes, #ribosomal genes
hb.genes <- c("HBA1","HBA2","HBB","HBG2","HBG1","HBD","HBE1","HBZ")#haemoglobin genes, there are no hb genes in the DEG list
mt.genes <- grep("^MT-", RNAlistRDS005_df$gene, value=TRUE)#mitochondrial genes, no genes
rb.genes <- grep(pattern = "^RP[SL]", RNAlistRDS005_df$gene, value=TRUE)#ribosomal genes, 165 genes
RNAlistRDS005_df <- RNAlistRDS005_df %>%
  dplyr::filter(!gene %in% rb.genes) #Focus on protein coding genes
##Rename cluster_id to major cell types
meta <- readRDS("/home/common/data/output/projects/ENDO/E044/A033/dataA003A033_metadata.rds")
meta$Symphony_Refined[meta$Symphony_Refined == "M$\\Phi$1-LYVE1"] <- "M-Phi-1-LYVE1"
meta$Symphony_Refined[meta$Symphony_Refined == "glandular early-secretory"] <- "glandular-early-secretory"
meta <- meta %>% 
  select(Symphony_Global_main5,Symphony_Refined,Symphony_Refined_A029_United) %>%
  distinct()
RNAlistRDS005_df <- RNAlistRDS005_df %>%
  mutate(Symphony_Global_main5 = cluster_id)
RNAlistRDS005_df$Symphony_Global_main5 <- case_when(
  RNAlistRDS005_df$Symphony_Global_main5 %in% meta$Symphony_Refined ~ meta$Symphony_Global_main5[match(RNAlistRDS005_df$Symphony_Global_main5, meta$Symphony_Refined)],
  RNAlistRDS005_df$Symphony_Global_main5 %in% meta$Symphony_Refined_A029_United ~ meta$Symphony_Global_main5[match(RNAlistRDS005_df$Symphony_Global_main5, meta$Symphony_Refined_A029_United)],
  TRUE ~ RNAlistRDS005_df$Symphony_Global_main5)
RNAlistRDS005_df$Symphony_Global_main5[RNAlistRDS005_df$Symphony_Global_main5 == "myeloid" |
                                      RNAlistRDS005_df$Symphony_Global_main5 == "lymphocyte"] <- "immune"

#Barplot unique genes per main cell type
RNAlistRDS005_df %>% 
  select(Symphony_Global_main5, gene, stage, cluster_id) %>%
  ggplot(aes(x = stage, fill = factor(Symphony_Global_main5, levels = c("immune", "endothelial", "epithelial", "stromal")))) +
  geom_bar() +
  theme(axis.text.x = element_text(angle = 45, vjust = 0.5, hjust = 0.3))

#Barplot unique genes per main cell type, filter for duplicates. Doublecheck numbers
RNAlistRDS005_df %>% 
  select(Symphony_Global_main5, gene, stage) %>%
  distinct() %>%
  ggplot(aes(x = stage, fill = factor(Symphony_Global_main5, levels = c("immune", "endothelial", "epithelial", "stromal")))) +
  geom_bar() +
  theme(axis.text.x = element_text(angle = 45, vjust = 0.5, hjust = 0.3))
ggsave(paste0(dir_out,"BarPlot_DEGs_main4.pdf"), 
       plot = last_plot(), 
       height = 4, width = 10)

RNAlistRDS005_df %>% 
      select(Symphony_Global_main5, gene, stage) %>%
      distinct() %>%
      count(stage, Symphony_Global_main5)

RNAlistRDS005_df %>% 
      filter(logFC >= 0.5) %>%
      distinct(gene, stage) %>%
      count(stage)

RNAlistRDS005_df %>% 
      select(gene, stage) %>%
      distinct() %>%
      count(stage)

table <- RNAlistRDS005_df %>% 
      select(Symphony_Global_main5, gene, stage) %>%
      distinct() %>%
      count(Symphony_Global_main5, stage)

```

Supplementary Data Fig. 7a
```{r cycling fibroblasts}
#LTs annotation for cycling fibroblasts
data <- readRDS("/home/common/data/output/projects/ENDO/E031/A004/proliferative_SCT_percent.mt_sample_integration_batch_hvg_annotated.RDS")
meta <- data@meta.data
rm(data)
gc()
table(meta$Annotation, meta$EndometriosisStatus)

#Make cell id compatible with new data set
meta$cell_id <- paste(meta$sample, meta$cell.names, sep = '_')
rownames(meta) <- meta$cell_id

#Load entire dataset and add LTs annotation
data_ds <- readRDS("/home/common/data/output/projects/ENDO/E044/A033/dataA003A033_downsampled.rds")
data_ds <- AddMetaData(data_ds,meta[,c("Annotation", "scds.cxds.score")])
DimPlot(data_ds_new,group.by = "Annotation")

#Highlight cycling fibroblasts
Idents(data_ds) <- data_ds$Annotation
table(data_ds$Annotation)
cyclingfibs <- WhichCells(data_ds, idents = c("Cycling Fibroblasts"))
DimPlot(data_ds, reduction = "umap", cells.highlight = cyclingfibs, cols.highlight = c("darkred"), cols= "grey", raster = FALSE)


```

Figure 4e
```{r Barplots for  filter response median rank E042A027}
#Load metadata
meta_final <- readRDS("/home/common/data/output/projects/ENDO/E044/A074/dataA003A033_metadata_ptime_snn_Final.rds")

#Set variables to loop over
DIMRED <- c("corrgpca","pca")
ANNOTATION <- c("Symphony_Refined","AnnotationMain")
MEDIAN <- c("median_rank_endo_logic003","median_rank_control_logic003","median_rank_endo_logic005","median_rank_control_logic005","median_rank_endo_logic01","median_rank_control_logic01")

for (dim in DIMRED) {
#Read filter ranking
average_filter_ranking <- readRDS(paste0("/home/common/data/output/projects/ENDO/E042/A027/response_cellcnn_manually/all_cells/",dim,"_average_filter_ranking.rds"))

#Combine tables
meta <- left_join(meta_final, average_filter_ranking %>% select(cell_ident, median_rank_endo, median_rank_control), by = "cell_ident") 

#Exclude NA
metaNAfree <- meta %>%
  select(median_rank_endo, median_rank_control,Symphony_Refined,AnnotationMain) %>%
  na.omit() %>%
  filter(Symphony_Refined != "granulocyte")

endo_threshold003 <- quantile(metaNAfree$median_rank_endo, 0.03)
control_threshold003 <- quantile(metaNAfree$median_rank_control, 0.03)
endo_threshold005 <- quantile(metaNAfree$median_rank_endo, 0.05)
control_threshold005 <- quantile(metaNAfree$median_rank_control, 0.05)
endo_threshold01 <- quantile(metaNAfree$median_rank_endo, 0.1)
control_threshold01 <- quantile(metaNAfree$median_rank_control, 0.1)

# Create the new column with ranking logic threshold
metaNAfree <- metaNAfree %>%
  mutate(median_rank_endo_logic003 = median_rank_endo <= endo_threshold003) %>%
  mutate(median_rank_control_logic003 = median_rank_control <= control_threshold003) %>%
  mutate(median_rank_endo_logic005 = median_rank_endo <= endo_threshold005) %>%
  mutate(median_rank_control_logic005 = median_rank_control <= control_threshold005) %>%
  mutate(median_rank_endo_logic01 = median_rank_endo <= endo_threshold01) %>%
  mutate(median_rank_control_logic01 = median_rank_control <= control_threshold01) 

# Create bar plots
dir.create(paste0(dir_out,"median_rank_plots/"))

for (ann in ANNOTATION) {
  for (med in MEDIAN) {
    # Create plot
    plot1 <- ggplot(metaNAfree, aes_string(x = ann)) +
      geom_bar(position = "identity") +
      scale_y_continuous(trans = "log10") +
      theme_minimal() +
      theme(axis.text.x = element_text(angle = 90, hjust = 1)) +
      ggtitle(paste0(dim, "___",ann, "___", med))
    plot2 <- ggplot(metaNAfree, aes_string(x = ann, fill = med)) +
      geom_bar(position = "fill") +
      theme_minimal() +
      theme(axis.text.x = element_text(angle = 90, hjust = 1))
    combined_plot <- plot1 / plot2
    # Save plot
    ggsave(filename = paste0(dir_out, "/median_rank_plots/", dim, "___",ann, "___", med, ".pdf"), plot = combined_plot, width = 13, height = 10)
  }
}
}

```

```{r UMAPs for  filter response median rank E042A027}
#Load data and metadata
data <- readRDS("/home/common/data/output/projects/ENDO/E044/A033/dataA003A033.rds")
#data <- readRDS("/home/common/data/output/projects/ENDO/E044/A033/dataA003A033_downsampled.rds")
meta_final <- readRDS("/home/common/data/output/projects/ENDO/E044/A074/dataA003A033_metadata_ptime_snn_Final.rds")
average_filter_ranking <- readRDS(paste0("/home/common/data/output/projects/ENDO/E042/A027/response_cellcnn_manually/all_cells/pca_average_filter_ranking.rds"))
#Combine meta, clean up and do ranking
meta <- left_join(meta_final, average_filter_ranking %>% select(cell_ident, median_rank_endo, median_rank_control), by = "cell_ident") 
#Exclude NA
metaNAfree <- meta %>%
  select(median_rank_endo, median_rank_control,Symphony_Refined,AnnotationMain,cell_ident) %>%
  na.omit()
#Ranking
endo_threshold01 <- quantile(metaNAfree$median_rank_endo, 0.1)
control_threshold01 <- quantile(metaNAfree$median_rank_control, 0.1)
metaNAfree <- metaNAfree %>%
  mutate(median_rank_endo_logic01 = median_rank_endo <= endo_threshold01) %>%
  mutate(median_rank_control_logic01 = median_rank_control <= control_threshold01) %>%
  select(Symphony_Refined, median_rank_endo_logic01, median_rank_control_logic01,cell_ident)
#Make summary table
summary_table <- metaNAfree %>%
  select(Symphony_Refined, median_rank_endo_logic01, median_rank_control_logic01) %>%
  group_by(Symphony_Refined) %>%
  summarize(
    percent_endo_TRUE = mean(median_rank_endo_logic01) * 100,
    percent_control_TRUE = mean(median_rank_control_logic01) * 100
  )
write.csv(summary_table, paste0(dir_out,"summary_pca_average_filter_ranking_10perc.csv"))

#Highlight 10% cells
endo_true_cells <- subset(metaNAfree, median_rank_endo_logic01 == TRUE)$cell_ident
control_true_cells <- subset(metaNAfree, median_rank_control_logic01 == TRUE)$cell_ident
p1 <- DimPlot(data, reduction = "umap", cells.highlight = endo_true_cells, cols.highlight = c("darkred"), cols= "grey", raster = FALSE) + ggtitle("endo_true_cells")
p2 <- DimPlot(data, reduction = "umap", cells.highlight = control_true_cells, cols.highlight = c("darkred"), cols= "grey", raster = FALSE) + ggtitle("control_true_cells")






```


make DEG table for paper
```{r}
#RNA assay DEGs from proliferative samples
RNAlistRDS005_df <- read.csv("/home/common/data/output/projects/ENDO/E044/A031/RNAlistRDS005_unmerged.csv") %>%
  filter(annotation_reference == "Tan") %>%
  filter(AveExpr > 18.53202)
##Exclude Haemoglobin genes, mitochondrial genes, #ribosomal genes
hb.genes <- c("HBA1","HBA2","HBB","HBG2","HBG1","HBD","HBE1","HBZ")#haemoglobin genes, there are no hb genes in the DEG list
RNAlistRDS005_df[RNAlistRDS005_df$gene %in% hb.genes, ]#no hb genes
#mitochondrial genes
mt.genes <- grep("^MT-", RNAlistRDS005_df$gene, value=TRUE)#mitochondrial genes, no genes
rb.genes <- grep(pattern = "^RP[SL]", RNAlistRDS005_df$gene, value=TRUE)#ribosomal genes, 77 genes
RNAlistRDS005_df <- RNAlistRDS005_df %>%
  dplyr::filter(!gene %in% rb.genes)

write.csv(RNAlistRDS005_df, paste0(dir_out,"RNAlistRDS005_AveExpr18.53202_noRBgenes.csv"))

length(unique(RNAlistRDS005_df$gene)) #427
```
